src/helpers.cc

```diff
@@ -312,10 +296,17 @@ LiteClient::LiteClient(Config& config_in)
   KeyManager keys(storage, config.keymanagerConfig());
   keys.copyCertsToCurl(*http_client);
 
-  registerComposeAppManager(ostree_sysroot);
-
-  primary =
-      std::make_shared<SotaUptaneClient>(config, storage, http_client, nullptr, primary_ecu.first, primary_ecu.second);
+  // TODO: consider improving this factory method
+  if (config.pacman.type == ComposeAppManager::Name) {
```

Comment:
I think you'll need an `ifdef BUILD_DOCKERAPP` around this, or some other cleaner way.

Reply:
what about compiling everything and just let choose a package manager type by using the configuration variable. Effectively,  lmp-device-register could drive it.

Reply:
+100